src/http-adapter/jquery/index.js

```diff
@@ -0,0 +1,84 @@
+/***
+ * @module http-adapter/jquery
+ */
+import HttpAdapter, { HTTP_METHOD } from '../index'
+import { ajax } from 'jquery'
+
+export default class JqueryHttpAdapter extends HttpAdapter {
+  /**
+   * The Http request method implemented by jQuery adapter class.
+   * @param {Object} options - request options to be propagated to the request
+   * @param {string} options.url - the url of the request
+   * @param {HTTP_METHOD} options.method - the method of the request
+   * @param {Object} options.headers - A key/value object of headers to be sent with the request
+   * @param {Object} options.data - A key/value payload to be sent with the request. If method is GET, these
+   * options will be transformed into query paramters
+   * @param {boolean} options.isInternal - Whether a request is being sent to an internal server
+   * or not to invoke request/response interceptors
+   * @returns {Promise} Promise relating to request resolution.
+   */
+  async _makeRequest ({
+    url,
+    method,
+    headers,
+    data,
+    isInternal
+  }) {
+    // Stringify data before any async stuff, just in case it's accidentally a mutable object (e.g.
+    // some instrumented Vue data)
+    if (data && [HTTP_METHOD.PATCH, HTTP_METHOD.POST].indexOf(method) > -1) {
+      data = JSON.stringify(data)
+    }
+
+    const requestInterceptor = this.requestInterceptor
+    const responseInterceptor = this.responseInterceptor
+
+    return new Promise((resolve, reject) => {
+      let request = {
+        url,
+        type: method,
+        headers,
+        beforeSend (xhr, options) {
+          if (isInternal) {
```

Comment:
What would be the implications of just defaulting to always use the interceptors, or alternatively checking if they have been set and always using them if they were set? Put another way, what is the use case for toggling them off (via `isInternal` or a hypothetical `useInterceptors`). if they have already been set?

Reply:
Bill talked to me about this via Slack,
> Potentially if the requests are going to a third party, which many of them do. Or, we just dont use the request method and use something else. we dont want to be sending header data to those API as it is a security issue. We could also just use the interceptors and check the domain on the interceptor and if its not our url prefix / host, bypass it

I’m fine with either way (using `useInterceptors` flag or making the app responsible for checking whether to apply the logic within the interceptors).

We could also think about whether to permit request-level interceptors and toggle adapter-level interceptors off by default for one-off requests, but I may be overengineering. I'm good with the simplest improvement and making more changes as necessary when use cases reveal themselves.

Reply:
> Bill talked to me about this via Slack,
> 
> > Potentially if the requests are going to a third party, which many of them do. Or, we just dont use the request method and use something else. we dont want to be sending header data to those API as it is a security issue. We could also just use the interceptors and check the domain on the interceptor and if its not our url prefix / host, bypass it
> 
> I’m fine with either way (using `useInterceptors` flag or making the app responsible for checking whether to apply the logic within the interceptors).
> 
> We could also think about whether to permit request-level interceptors and toggle adapter-level interceptors off by default for one-off requests, but I may be overengineering. I'm good with the simplest improvement and making more changes as necessary when use cases reveal themselves.

@electrachong To just keep it simple, I think making the applications responsible for this is the easiest option. This means removing `isInternal` from the API. Will update shortly 🙂